docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,192 @@
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+import {Order, Customer} from '../models';
+import {OrderRepository} from './order.repository.ts';
+import {
+  DefaultCrudRepository,
+  juggler,
+  HasManyRepositoryFactory,
+} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.create))
+- `find` finding target model instance(s) belonging to customer model instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.find))
+- `delete` for deleting target model instance(s) belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.delete))
+- `patch` for patching target model instance(s) belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.patch))
+
+## Using hasMany constrained repository in a controller
+
+The same pattern used for ordinary repositories to expose their CRUD APIs in
+controller methods is employed for `hasMany` repositories. Once the hasMany
+relation has been defined and configured, controller methods can call the
+underlying constrained repository CRUD APIs and expose them as routes once
+decorated with
+[Route decorators](Routes.md#using-route-decorators-with-controller-methods). It
+will require the value of the foreign key and, depending on the request method,
+a partial value for the target model instance as demonstrated below.
+
+{% include code-caption.html
+content="src/controllers/customer-orders.controller.ts" %}
+
+```ts
+import {post, param, requestBody} from '@loopback/rest';
+import {customerRepository} from '../repositories/customer.repository.ts';
+import {Customer, Order} from '../models/';
+
+export class CustomerOrdersController {
+  constructor(
+    @repository(CustomerRepository)
+    protected customerRepository: CustomerRepository,
+  ) {}
+
+  @post('/customers/{id}/order')
+  async createCustomerOrders(
```

Comment:
Perhaps `createOrderOfCustomer` would be a better name, considering that the first method argument is `customerId`?

If not, then at least change the name into singular because we are creating a single order: `createCustomerOrder`.

Another name to consider: `createOrder` or simply `create` - both the source and target model names are already captures in the controller name.

Reply:
Thank you for those suggestions @bajtos. I like either `createOrderOfCustomer` or `createOrder`. Let's go with `createOrder`.